### ISSUE TITLE
Use validation set when selecting the best epoch

### ISSUE THREAD

**Issue by gwinndr** [gwinndr/MusicTransformer-Pytorch#10]
Need to select the best epoch based on the validation set and use that best epoch to get the final accuracy and loss.

Using the test set to select the best epoch, biases the model towards the test set. The test set is supposed to represent "in the wild" data and evaluate how well the final model performs on yet-to-be-seen data.
